IDinsight/aaq-core
# Multilingual/edit

Reviewer: @sidravi1 

Estimate: 1h00

---

## Ticket

Fixes: [AAQ-428](https://idinsight.atlassian.net/browse/AAQ-428)

## Description, Motivation and Context

### Goal
The goal of this PR is to add a feature to edit multilingual content. 
### Changes
What changed on the backend?
- Added a route to edit content. The endpoint will edit an existing language version of a content and will also add a new language version to an existing text if the language version is not existing. 
-What changed on the frontend:
- Updated the edit page to add new language version to a content. 
- Added a Modal to edit a new language version or go to landing page after succesful edit 
## How has this been tested?
Endpoint has been tested on the backend. And frontend has been tested as well to make sure it is working as intended. Test cases will be added later. No need to add a language as english is added as a default language by default in the migrations.
### How to test
To test, add a few contents using the API (preferably in different languages) login to the frontend app and try updating. Both using the ContentBox and the read content modal. What to test?
- Test that each language version is updated as it should.
-  Test that you can add a new language version using edit page.
- Test that you redirected to the language chosen when viewing. 

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant



[AAQ-428]: https://idinsight.atlassian.net/browse/AAQ-428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
